File: game-server/src/main/java/work/fking/pangya/game/packet/outbound/RareShopOpenResponsePacket.java
package work.fking.pangya.game.packet.outbound;

import io.netty.buffer.ByteBuf;
import work.fking.pangya.networking.protocol.OutboundPacket;

public class RareShopOpenResponsePacket implements OutboundPacket {

    private static final int ID = 0x010B;

    @Override
    public void encode(ByteBuf target) {
        target.writeShortLE(ID);
        target.writeIntLE(0xFFFFFFFF);
        target.writeIntLE(0xFFFFFFFF);
        target.writeIntLE(0);
    }
}

File: game-server/src/main/kotlin/work/fking/pangya/game/packet/outbound/RareShopOpenResponsePacket.kt
package work.fking.pangya.game.packet.outbound

import io.netty.buffer.ByteBuf
import work.fking.pangya.networking.protocol.OutboundPacket

class RareShopOpenResponsePacket : OutboundPacket {

    override fun encode(buffer: ByteBuf) {
        buffer.writeShortLE(0x10B)
        buffer.writeIntLE(-0x1)
        buffer.writeIntLE(-0x1)
        buffer.writeIntLE(0)
    }
}
